sdk/identity/azure-identity/src/azure_pipelines_credential.cpp

```diff
@@ -61,26 +38,21 @@ AzurePipelinesCredential::AzurePipelinesCredential(
     AzurePipelinesCredentialOptions const& options)
     : TokenCredential("AzurePipelinesCredential"), m_serviceConnectionId(serviceConnectionId),
       m_systemAccessToken(systemAccessToken),
-      m_clientCredentialCore(tenantId, options.AuthorityHost, options.AdditionallyAllowedTenants),
       m_httpPipeline(HttpPipeline(options, "identity", PackageVersion::ToString(), {}, {}))
 {
   m_oidcRequestUrl = _detail::DefaultOptionValues::GetOidcRequestUrl();
+  ClientAssertionCredentialOptions clientAssertionCredentialOptions{};
```

Comment:
This is why SimulateTokenRequest does not work for you in the tests - you use default transport options, which means trasport options are initialized with WinHTTP or libcurl. Instead, you should forward them from TokenCredentialOptions (options). I.e. You should add a constructor to ClientAssertionCredential, and takes base class of TokenCredentialOptions, from which all the specific credential options are derived. And everything extra that ClientAssertionCredentialOptions adds on top of the TokenCredentialOptions it inherits, go via separate constructor parameters.
We use this pattern in EnvironmentCredential - see how it initializes different types of cfredentials with its own basic TokenCredentialOPtions + extra parans for each cred type.